### ISSUE TITLE
IJ plugin: IDE misses some variable validation errors

### ISSUE THREAD

**Issue by martinbonnin** [apollographql/apollo-kotlin#6219]
### Version

4.0.1

### Summary

This query using the internal platform API:

```graphql
query GetSparklines($graph: String!){
  service(id: $graph) {
    today: statsWindow(from: "-86400", to: "-0", resolution: R15M) {
      queryStats {
        timestamp
        metrics {
          totalRequestCount
          requestsWithErrorsCount
        }
      }
    }
  }
}
```

Fails validation with

```
> A failure occurred while executing com.apollographql.apollo.gradle.internal.GenerateSources
   > e: /Users/martinbonnin/git/studio-companion-app/android/app/src/main/graphql/apollo-platform/internal/GetSparklines.graphql: (2, 15): Variable `graph` of type `String!` used in position expecting type `ID!`
     ----------------------------------------------------
     [1]:query GetSparklines($graph: String!){
     [2]:  service(id: $graph) {
     [3]:    today: statsWindow(from: "-86400", to: "-0", resolution: R15M) {
     ----------------------------------------------------
```

But no error is shown in the editor

![Screenshot 2024-10-24 at 17 12 10](https://github.com/user-attachments/assets/2c751660-5bbf-4292-9032-4032815ed467)

**Comment by BoD** [comment]
I couldn't repro at first

<img width="591" alt="Screenshot 2024-10-29 at 11 35 13" src="https://github.com/user-attachments/assets/3904b7c6-fcec-422f-9ddc-aca97a8c7a07">

After disabling 'Contribute configuration to the GraphQL plugin' - which allows the GraphQL plugin to know where is the schema for a given operation, I could repro.

When the conf is not contributed, the GraphQL plugin defaults to using all schemas (so internal API + public API), which results in lots of schema validation errors because types have the same name. Because of this, some operation validations can't be done.

- The conf is contributed after a successful Gradle sync, and may take a while the first time (it is then cached into the project settings). During this time, this can happen.
- A gradle sync is not always automatically done when opening  project (¯\\\_(ツ)\_/¯).
- I've also witnessed intermittent issues where the conf was not contributed, and either forcing a gradle sync or re-opening the project fixed it... Unfortunately I never got to the bottom of it, couldn't find repro steps :(

If you can still repro after a Gradle sync, something else is wrong.

**Comment by martinbonnin** [comment]
> When the conf is not contributed, the GraphQL plugin defaults to using all schemas

Can we disable that/force the configuration to be always contributed? 

I'm okay with no highlighting at all but wrong highlitghting seems quite counter intuitive?

**Comment by BoD** [comment]
Agreed that the default behavior can be worse than no knowledge of the schema at all - #6228 should make this better.

**Comment by BoD** [comment]
Closed by #6228
